### PR TITLE
[Exam CSS Fix Continuation : ProblemStatement vs Instructions] Fix for Assesment Instructions

### PR DESCRIPTION
**Pull request by Malyuk-A**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/).
- [x] Client: I added multiple screenshots/screencasts of my UI changes

### Motivation and Context

This is a follow-up PR to [#3740](https://github.com/ls1intum/Artemis/pull/3740).
Look at @MDK5 [screenshots](https://github.com/ls1intum/Artemis/pull/3740#pullrequestreview-709056904) to see that the assessment instructions look different for almost every exercise type.
This PR fixes the mentioned issues and uses a homogeneous look similar to the problem statement header for students.

### Description

Look at the screenshots.
Also fixes inconsistency that "Problem Statement" for modeling exercises was bold and had a ":".

_The reason for so many commits is due to the other CSS PR's code about programming exercises. We merged the code into this branch to test its functionality. The current problem here is, that the same html/css is used for students and tutors(assesment), therefore it is not "instructions" but "problem statement". This will be fixed in a future refactoring._

### Steps for Testing
Check every exercise type assessment and look at the instruction header.
Also do not forget to minimize it and check if it is still correct and in line with the others.
Check if the same holds true for exam assessments.

### Screenshots
modeling | text | file-upload
:-----------:|:-----------:|:-----------:
  |   |

### CODE REVIEW

**Review by joschahenningsen — APPROVED** [review]
Looks good and code changes seem reasonable :+1:

**Review by katjanakosic — APPROVED** [review]
Works as expected on ts1 👍

File: src/main/webapp/app/exercises/programming/shared/code-editor/instructions/code-editor-instructions.component.html [hunk 1]
```diff
@@ -1,12 +1,19 @@
-<div id="cardInstructions" class="instructions-wrapper card border border-lightgrey">
-    <div class="card-header d-flex justify-content-center text-white bg-primary" (click)="toggleEditorCollapse($event)">
+<div id="cardInstructions" class="instructions-wrapper card collapsable-card" [ngStyle]="collapsed ? { height: '250px' } : { height: '100%' }">
```
Comment (MerlinMehmed) [inline src/main/webapp/app/exercises/programming/shared/code-editor/instructions/code-editor-instructions.component.html:1]: I know it's just one style option in the ngStyle clause but it's still inline css which is a bad practice. You can consider moving the height options to classes.

File: src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.scss [hunk 2]
```diff
@@ -162,17 +162,29 @@ Collapsable styling
...
+        justify-content: space-between;
+        flex-flow: column nowrap;
+        cursor: pointer;
+
+        .header {
+            writing-mode: vertical-rl;
+            margin-left: -3px;
+        }
+
+        fa-icon {
```
Comment (MerlinMehmed) [inline src/main/webapp/app/exercises/programming/shared/code-editor/layout/code-editor-grid.scss:182]: I am not really sure what you can do to resolve the Codacy issue. Maybe you can extract the fa-icon outside the .collapsed-instructions-card or add an id or a css class to the fa-icon in the HTML part.

**Review by MerlinMehmed — APPROVED** [review]
Works as expected, code changes look reasonable as well. I have added two minor comments which you can consider applying.

**Review by maximilianbergmann — APPROVED** [review]
Tested on TS3

File: src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.scss [hunk 3]
```diff
@@ -13,3 +13,7 @@ jhi-modeling-assessment {
 .editor-large {
     min-height: 100vh;
 }
+
+#problem-statement {
+    font-weight: normal !important;
+}
```
Comment (iozbeyli) [inline src/main/webapp/app/exercises/modeling/participate/modeling-submission.component.scss:19]: You are removing the strong tag from translation too, is it necessary to do the both?

**Pull request merged**